python/pyspark/sql/session.py

```diff
@@ -121,6 +121,19 @@ def newSession(self):
         """
         return self.__class__(self._sc, self._jsparkSession.newSession())
 
+    @property
+    @since(2.0)
+    def conf(self):
+        """Runtime configuration interface for Spark.
+
+        This is the interface through which the user can get and set all Spark and Hadoop
+        configurations that are relevant to Spark SQL. When getting the value of a config,
+        this defaults to the value set in the underlying :class:`SparkContext`, if any.
+        """
+        if not hasattr(self, "_conf"):
+            self._conf = RuntimeConfig(self._jsparkSession.conf())
+        return self._conf
+
     @since(2.0)
     def setConf(self, key, value):
```

Comment:
actually i was looking at this. initially runtimeconf has a lot of functions, but now it is pretty simple and no longer have a lot. I'm wondering if we should just remove that ... and have only SparkSession.setConf/getConf